### ISSUE TITLE
Support for targeted attacks for Universal Perturbation

### ISSUE THREAD

**Issue by hkakitani** [Trusted-AI/adversarial-robustness-toolbox#393]
**Is your feature request related to a problem? Please describe.**
Universal perturbation can only use untargeted attacks if it is using an attack that requires the target labels to be passed as an argument to their `generate()` function (ie. PGD, Carlini-Wagner).  When UP calls an attack's `generate()` function, it does not pass in the optional y argument that specifies target labels (https://github.com/IBM/adversarial-robustness-toolbox/blob/master/art/attacks/evasion/universal_perturbation.py#L134). 
**Describe the solution you'd like**
Requesting an option to specify the optional y argument for certain attacks that require a list of target labels in their `generate()` function.

**Comment by beat-buesser** [comment]
Hi @hkakitani Thank you very much for raising this issue! We will investigate and fix it as soon as possible.

**Comment by hkthirano** [comment]
We proposed a simple iterative method to generate UAPs for targeted attacks. Our method combines the simple iterative method for generating non-targeted UAPs and the fast gradient sign method for generating a targeted adversarial perturbation for an input.

https://github.com/hkthirano/targeted_UAP_CIFAR10

https://github.com/hkthirano/adversarial-robustness-toolbox/commit/d300f41cdc124be1cb25c0a1b7504dd98f18ab7f

Note that our algorithm was implemented using ART version 1.0.
